pkg/apis/pipeline/v1alpha1/pipelinerun_types_test.go

```diff
@@ -281,3 +281,106 @@ func TestPipelineRunGetServiceAccountName(t *testing.T) {
 		}
 	}
 }
+
+func TestPipelineRunGetPodSpecSABackcompatibility(t *testing.T) {
+	for _, tt := range []struct {
+		name    string
+		pr      *v1alpha1.PipelineRun
+		saNames map[string]string
+	}{
+		{
+			name: "test backward compatibility",
+			pr: &v1alpha1.PipelineRun{
+				ObjectMeta: metav1.ObjectMeta{Name: "pr"},
+				Spec: v1alpha1.PipelineRunSpec{
+					PipelineRef:        &v1alpha1.PipelineRef{Name: "prs"},
+					ServiceAccountName: "defaultSA",
+					ServiceAccountNames: []v1alpha1.PipelineRunSpecServiceAccountName{{
+						TaskName: "taskName", ServiceAccountName: "taskSA",
+					}},
+					TaskRunSpecs: []v1alpha1.PipelineTaskRunSpec{{
+						PipelineTaskName:       "taskName",
+						TaskServiceAccountName: "newTaskSA",
+					}},
+				},
+			},
+			saNames: map[string]string{
+				"unknown":  "defaultSA",
+				"taskName": "newTaskSA",
+			},
+		},
+		{
+			name: "mixed default SA backward compatibility",
+			pr: &v1alpha1.PipelineRun{
+				ObjectMeta: metav1.ObjectMeta{Name: "pr"},
+				Spec: v1alpha1.PipelineRunSpec{
+					PipelineRef:        &v1alpha1.PipelineRef{Name: "prs"},
+					ServiceAccountName: "defaultSA",
+					TaskRunSpecs: []v1alpha1.PipelineTaskRunSpec{{
+						PipelineTaskName:       "taskNameOne",
+						TaskServiceAccountName: "TaskSAOne",
+					}, {
+						PipelineTaskName:       "taskNameTwo",
+						TaskServiceAccountName: "newTaskTwo",
+					}},
+				},
+			},
+			saNames: map[string]string{
+				"unknown":     "defaultSA",
+				"taskNameOne": "TaskSAOne",
+				"taskNameTwo": "newTaskTwo",
+			},
+		},
+	} {
+		for taskName, expected := range tt.saNames {
+			sa, _ := tt.pr.GetTaskRunSpecs(taskName)
+			if expected != sa {
+				t.Errorf("%s: wrong service account: got: %v, want: %v", tt.name, sa, expected)
+			}
+		}
+	}
+}
+
+func TestPipelineRunGetPodSpec(t *testing.T) {
+	for _, tt := range []struct {
+		name         string
+		pr           *v1alpha1.PipelineRun
+		podTemplates map[string][]string
```

Comment:
minor nit: same as above -- add a comment/make field name clearer that this is the expected output from the test.